OpenRA.Mods.Common/Graphics/ContrailRenderable.cs

```diff
@@ -63,8 +65,8 @@ public IRenderable OffsetBy(in WVec vec)
 		public IFinalizedRenderable PrepareRender(WorldRenderer wr) { return this; }
 		public void Render(WorldRenderer wr)
 		{
-			// Need at least 4 points to smooth the contrail over
-			if (length - skip < 4)
+			var renderLength = length - skip;
+			if (renderLength <= 1)
```

Comment:
> fixed, but we may need to comment it here about why it must >=1 (and I don't know how to explain on this)

What about
```suggestion
			// Require at least two points to draw a tail
			var renderLength = length - skip;
			if (renderLength <= 1)
```
?

Reply:
can be but it is better we explain contrail length is actually the number of points drawing by tick, it is not designed as what it looks like.

Reply:
// HACK: The length of contrail is now acutally the number of the points to draw the contrail
			// and we require at least two points to draw a tail

Reply:
Why do you think this is a HACK?

Reply:
ummm, may be not the HACK.

Reply:
Could I use Note instead?

Reply:
I don't think either is necessary, but if you insist I'd prefer Note over HACK.

Reply:
ok, then I will use Note here

Reply:
fixed